airflow/utils/log/file_task_handler.py

```diff
@@ -19,18 +19,17 @@
 import logging
 import os
 from pathlib import Path
-from typing import TYPE_CHECKING, Optional
+from typing import Optional
 
 import httpx
 from itsdangerous import TimedJSONWebSignatureSerializer
 
 from airflow.configuration import AirflowConfigException, conf
+from airflow.models import TaskInstance
 from airflow.utils.context import Context
 from airflow.utils.helpers import parse_template_string, render_template_to_string
 from airflow.utils.log.non_caching_file_handler import NonCachingFileHandler
-
-if TYPE_CHECKING:
-    from airflow.models import TaskInstance
```

Comment:
Better to keep this conditional to avoid potential circuliar import. `airflow.models` imports _way_ too many things it’s not a good idea to import it in `utils`.